core/domain/collection_domain.py

```diff
@@ -230,13 +272,49 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id)
 
 
+class CollectionDict(TypedDict):
+    """Dictionary representing the Collection object."""
+
+    id: str
+    title: str
+    category: str
+    objective: str
+    language_code: str
+    tags: List[str]
+    schema_version: int
+    nodes: List[CollectionNodeDict]
+    version: int
+    created_on: Union[datetime.datetime, str]
+    last_updated: Union[datetime.datetime, str]
```

Comment:
https://github.com/oppia/oppia/blob/094f6e8c17c9710459b3f9a70aa0393380ec0b68/core/domain/collection_domain.py#L407
https://github.com/oppia/oppia/blob/094f6e8c17c9710459b3f9a70aa0393380ec0b68/core/domain/collection_domain.py#L411

However, we only need `created_on` and `last_updated` to be string in TypedDict, but as we updated the typedDict according to current domain object. we don't longer needed them to be string. Because previously we are storing string of datetime in Dict and then passing that dict to `json.dumps()` function.